clang/lib/Sema/SemaOverload.cpp

```diff
@@ -7976,7 +7976,8 @@ void Sema::AddConversionCandidate(
 
   if (Conversion->getTrailingRequiresClause()) {
     ConstraintSatisfaction Satisfaction;
-    if (CheckFunctionConstraints(Conversion, Satisfaction) ||
+    if (CheckFunctionConstraints(Conversion, Satisfaction, /*UsageLoc=*/{},
```

Comment:
This isn't really specific to lambdas right?  Doing this on a function seems like it would do the wrong thing, right?

Reply:
Hmm... Yes, the flag would tell the LSI to look into its parents as well, although we don't have any coverage test cases for it yet. Maybe I think we can narrow it down to the inside of CheckFunctionConstraints.